addons/api/addon/generated/models/host-catalog.js

```diff
@@ -55,10 +55,11 @@ export default class GeneratedHostCatalogModel extends BaseModel {
   })
   plugin;
 
-  // AWS & Azure
+  // AWS, Azure & GCP
 
   @attr('boolean', {
     for: 'plugin',
+    compositeType: ['aws', 'azure', 'gcp'],
```

Comment:
yeah you are right, I'm looking into refactoring the exisiting `for` in app serializer to handle this

Reply:
thanks for the feedback - I moved the compositeType logic to app serializer and handled the credentialType part in resource specific serializer since it is resource specific. Let me know what you think